src/java.base/share/classes/sun/security/provider/certpath/RevocationChecker.java

```diff
@@ -740,9 +746,37 @@ private void checkOCSP(X509Certificate cert,
                         null, -1);
                 }
 
+                List<Extension> tmpExtensions = new ArrayList<Extension>();
+                if (rp.ocspNonce) {
+                    if (nonce == null) {
+                        try {
+                            // create the 16-byte nonce by default
+                            Extension nonceExt = new OCSPNonceExtension(DEFAULT_NONCE_BYTES);
+                            tmpExtensions.add(nonceExt);
+
+                            if (debug != null) {
+                                debug.println("Default nonce has been created in the OCSP extensions");
+                            }
+                        } catch (IOException e) {
+                            throw new CertPathValidatorException("Failed to create the default nonce " +
+                                    "in OCSP entensions");
+                        }
+                    } else {
+                        throw new CertPathValidatorException("Application provided nonce cannot be " +
+                                "used if the value of the jdk.security.certpath.ocspNonce system " +
+                                "property is true");
+                    }
+                } else {
+                    if (nonce != null) {
+                        if (debug != null) {
+                            debug.println("Using application provided nonce");
+                        }
+                    }
+                }
+
                 response = OCSP.check(Collections.singletonList(certId),
                         responderURI, issuerInfo, responderCert, null,
-                        ocspExtensions, params.variant());
+                        rp.ocspNonce ? tmpExtensions : ocspExtensions, params.variant());
```

Comment:
Here you can just pass in `ocspExtensions` since it will contain the nonce if the property has been set.

Reply:
No change as tmpExtensions is needed.